Localization and FeatureGroup Enhancement

Tuesday1028 opened the pull request:
### **It seems to be working well, but it still needs a comprehensive test.**

---

#### Ability to generate and load translation files for feature names, descriptions and settings. **[OK]**

#### Translation files would contain a dictionary of identifiers which map to strings containing the translated text. **[OK]**

#### should be automatically generated based on settings property names
 - Feature.Name **[OK]**
 - Feature.Description **[OK]**
 - BoolSetting  **[OK]**
 - ButtonSetting **[OK]**
 - ColorSetting **[OK]**
 - EnumSetting **[OK]**
 - EnumListSetting **[OK]**
 - LabelSetting **[OK]**
 - etc **[OK]**
#### hot swappable, re-generate settings menu to switch language **[OK]**
#### automatically applied to mod settings menu by changing the translation files **[OK]**

Tuesday1028 commented:
I don't have any mono build, so I only added multi-language support for the IL2CPP build.

AuriRex commented:
Ohh, very nice - looks promising.

I will check it out whenever I have some time to do so 👍
(#54)

AuriRex commented on TheArchive.Core/Core/Localization/LocalizationCoreService.cs:
```diff
@@ -0,0 +1,140 @@
+
+        public static void Init()
+        {
+            string dir = string.Concat(Path.GetDirectoryName(ArchiveMod.CORE_PATH), $"\\Localization\\{(LoaderWrapper.IsGameIL2CPP() ? "IL2CPP" : "MONO")}\\");
```
We probably don't have to separate Mono from IL2CPP as both (should) have the same Features;  
I will have to test that tho
```suggestion
            string dir = string.Concat(Path.GetDirectoryName(ArchiveMod.CORE_PATH), $"\\Localization\\");
```

AuriRex replied:
saw your commit 5fc69f44904dab971473b65ea13cb57a978be3aa
fixes the second time this is used, original one in `LocalizationCoreService` is still here haha

AuriRex reviewed (commented):
Overall looks pretty good,

I am still unsure about the save paths of the localization files and if it might be better to use `Path.Combine(LocalFiles.SaveDirectoryPath, "Localization")` instead and just treat files next to the .dll as templates?  
Maybe include a `int version` in `FeatureLocalizationData` to force localization updates from templates?

If you have a better idea feel free to bring it up 👍

Tuesday1028 commented:
> Overall looks pretty good,
> 
> I am still unsure about the save paths of the localization files and if it might be better to use `Path.Combine(LocalFiles.SaveDirectoryPath, "Localization")` instead and just treat files next to the .dll as templates? Maybe include a `int version` in `FeatureLocalizationData` to force localization updates from templates?
> 
> If you have a better idea feel free to bring it up 👍

I think it is unnecessary to save the translated file to the settings directory. It is a complete waste of storage space. Loading the translation file from the "Localization" directory in the same directory as the DLL is sufficient.

I don't think the version of the translation file matters in practice. If the goal is to automatically modify the JSON structure when there are changes to the members of the "Feature" class, that has already been implemented in my new commits.

AuriRex commented:
Sorry for taking so long, I'll try to get this merged in the next few days probably haha

AuriRex reviewed (approved):
Okay,
I changed a few field name styles, moved a few methods around and did some small touchups that I could see.
Should be good, unless you have any objections to my changes; I'll be merging this tomorrow probably 👍

Tuesday1028 commented:
> Okay, I changed a few field name styles, moved a few methods around and did some small touchups that I could see. Should be good, unless you have any objections to my changes; I'll be merging this tomorrow probably 👍


There are some language-related modifications [here](https://github.com/Tuesday1028/GTFO_TheArchive/tree/main) that haven't been synchronized with the multi-lang branch. 
Multi-language is not yet perfect. FS in TheArchive.Core cannot be recognized.
Some members have missed FS attribute, causing the multi-language recognition to fail. Need double-check.

The main branch also includes the Enhanced FeatureGroup. 
It would be best if you could check Enhanced FeatureGroup.

Additionally, I haven't fully completed the Chinese translation. 

Translated by GPT-3.5
